tests/Generator/Resolver/Value/Chainable/FixtureMethodCallReferenceResolverTest.php

```diff
@@ -0,0 +1,215 @@
+     * @expectedExceptionMessage Expected method "Nelmio\Alice\Generator\Resolver\Value\Chainable\FixtureMethodCallReferenceResolver::resolve" to be called only if it has a resolver.
+     */
+    public function testCannotResolveValueIfHasNoResolver()
+    {
+        $value = new FixtureMethodCallValue(new FakeValue(), new FunctionCallValue('method'));
+        $resolver = new FixtureMethodCallReferenceResolver();
+        $resolver->resolve($value, new FakeFixture(), ResolvedFixtureSetFactory::create(), [], new GenerationContext());
+    }
+
+    public function testReturnsSetWithResolvedValue()
+    {
+        $value = new FixtureMethodCallValue(
+            $reference = new FakeValue(),
+            $functionCall = new FunctionCallValue('getFoo', [
+                $arg1 = new FakeValue(),
+            ])
+        );
+        $fixture = new FakeFixture();
+        $set = ResolvedFixtureSetFactory::create(new ParameterBag(['foo' => 'bar']));
+        $scope = ['val' => 'scopie'];
+        $context = new GenerationContext();
+        $context->markIsResolvingFixture('foo');
+
+        $valueResolverContext = new GenerationContext();
+        $valueResolverContext->markIsResolvingFixture('foo');
+        $valueResolverContext->markAsNeedsCompleteGeneration();
+
+        $dummyProphecy = $this->prophesize(DummyWithGetter::class);
+        $dummyProphecy->getFoo('resolved_argument')
+            ->shouldBeCalled()
+            ->willReturn('resolved_value');
+
+        $valueResolverProphecy = $this->prophesize(ValueResolverInterface::class);
+        $valueResolverProphecy
+            ->resolve($arg1, $fixture, $set, $scope, $context)
+            ->willReturn(
+                new ResolvedValueWithFixtureSet(
+                    'resolved_argument',
+                    $newSet = ResolvedFixtureSetFactory::create(new ParameterBag(['ping' => 'pong']))
+                )
+            )
+        ;
+        $valueResolverProphecy
+            ->resolve($reference, $fixture, $newSet, $scope, $valueResolverContext)
+            ->willReturn(
+                new ResolvedValueWithFixtureSet(
+                    $instance = $dummyProphecy->reveal(),
+                    $newSet
+                )
+            )
+        ;
+        /** @var ValueResolverInterface $valueResolver */
+        $valueResolver = $valueResolverProphecy->reveal();
+
+        $expected = new ResolvedValueWithFixtureSet('resolved_value', $newSet);
+
+        $resolver = new FixtureMethodCallReferenceResolver($valueResolver);
+        $actual = $resolver->resolve($value, $fixture, $set, $scope, $context);
+
+        $this->assertEquals($expected, $actual);
+
+        $valueResolverProphecy->resolve(Argument::cetera())->shouldHaveBeenCalledTimes(2);
+        $dummyProphecy->getFoo(Argument::cetera())->shouldHaveBeenCalledTimes(1);
+    }
+
+    public function testCatchesThrowablesToThrowResolverException()
```

Comment:
`testThrowsAResolverExceptionOnError`? (error to be taken in the english sense, not the `\Error` object)